content/en/tests/setup/go.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Go Tests
+code_lang: go
+type: multi-code-lang
+code_lang_weight: 60
+aliases:
+  - /continuous_integration/setup_tests/go
+  - /continuous_integration/tests/go
+  - /continuous_integration/tests/setup/go
+further_reading:
+    - link: "/continuous_integration/tests/containers/"
+      tag: "Documentation"
+      text: "Forwarding Environment Variables for Tests in Containers"
+    - link: "/continuous_integration/tests"
+      tag: "Documentation"
+      text: "Explore Test Results and Performance"
+    - link: "/tests/test_impact_analysis/go"
+      tag: "Documentation"
+      text: "Speed up your test jobs with Test Impact Analysis"
+    - link: "/continuous_integration/troubleshooting/"
+      tag: "Documentation"
+      text: "Troubleshooting CI Visibility"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">Test Optimization is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+## Compatibility
+
+Supported test frameworks:
+
+- `testing` package
+
+## Configuring reporting method
+
+To report test results to Datadog, you need to configure the Datadog Go library:
+
+{{< tabs >}}
+
+{{% tab "Cloud CI Provider" %}}
+{{% ci-agentless %}}
+{{% /tab %}}
+
+{{% tab "On-Premises CI Provider" %}}
+{{% ci-agent %}}
+{{% /tab %}}
+
+{{< /tabs >}}
+
+## Installing Orchestrion
+
+Orchestrion is a tool to process Go source code at compilation time and automatically insert instrumentation using `dd-trace-go`.
+
+You only need to install `orchestrion` from [https://github.com/datadog/orchestrion][1] using:
+
+```bash
+$ go install github.com/DataDog/orchestrion@latest
+```
+
+  - <details><summary>Optional: project <tt>go.mod</tt> registration</summary>
+
+    >  You can automatically add `orchestrion` to your project's dependencies by running:
+    > ```bash
+    > $ orchestrion pin
+    > ```
+    > This will:
+    > 1. Create a new `orchestrion.tool.go` file containing content similar to:
+    >     ```go
+    >     // Code generated by `orchestrion pin`; DO NOT EDIT.
+    >
+    >     // This file is generated by `orchestrion pin`, and is used to include a blank import of the
+    >     // orchestrion package(s) so that `go mod tidy` does not remove the requirements rom go.mod.
+    >     // This file should be checked into source control.
+    >
+    >     //go:build tools
+    >
+    >     package tools
+    >
+    >     import _ "github.com/DataDog/orchestrion"
+    >     ```
+    > 2. Run `go get github.com/DataDog/orchstrion@<current-release>` to make sure the project version corresponds to the
+    >    one currently being used
+    > 3. Run `go mod tidy` to make sure your `go.mod` and `go.sum` files are up-to-date
+    >
+    > If you do not run this command, it will be done automatically when required. Once done, the version of `orchestrion`
```

Comment:
```suggestion
    > If you do not run this command, it is done automatically when required. Once done, the version of `orchestrion`
```
Can you clarify the mechanism that runs the command automatically? I'm looking for something like "If you do not run this command, XXX runs it automatically when required."

Also, I don't understand what "Once done" means in this context. Does it mean after `orchestrion pin` is run?

Reply:
>Can you clarify the mechanism that runs the command automatically?
I don't know the details. This extracted directly from the orchestrion readme file. https://github.com/DataDog/orchestrion

>Also, I don't understand what "Once done" means in this context. Does it mean after orchestrion pin is run?
yes